staging/src/k8s.io/component-base/metrics/registry.go

```diff
@@ -49,6 +49,49 @@ func shouldHide(currentVersion *semver.Version, deprecatedVersion *semver.Versio
 	return false
 }
 
+func validateShowHiddenMetricsVersion(currentVersion semver.Version, targetVersionStr string) error {
+	if targetVersionStr == "" {
+		return nil
+	}
+
+	// We assume the input target version with the format `x.y`, here simply suffix `0` as patch version.
+	// It helps to compare with current version.
+	pathVersion := fmt.Sprintf("%s.0", targetVersionStr)
+
+	targetVersion, err := semver.Parse(pathVersion)
+	if err != nil {
+		return fmt.Errorf("specified --show-hidden-metrics-for-version '%v' not follows the version format x.y", targetVersionStr)
+	}
+
+	maxAllowedVersion, err := semver.Make(fmt.Sprintf("%d.%d.0", currentVersion.Major, currentVersion.Minor))
```

Comment:
I think you don't need any of this semver code. Only the previous minor version is allowed, so you can print it to a string and compare directly.

Reply:
I think that can handle the above error condition, too. The error message can say, `--show-hidden-metrics-for-version must be omitted or have the value "%v". Only the previous minor version is allowed.`